moviebot/agent/agent.py

```diff
@@ -125,29 +99,25 @@ def initialize(self, user_id=None):  # noqa: C901
         self.nlg = NLG(dict(ontology=self.ontology))
         data_config["slots"] = list(self.nlu.intents_checker.slot_values.keys())
 
-        if self.config.get("TELEGRAM", False):
-            self.isBot = True
-            # self.new_user = self.config['new_user'][user_id]
+        self.isBot = self.config.get("TELEGRAM", False)
 
         self.dialogue_manager = DialogueManager(
             data_config, self.isBot, self.new_user
         )
 
-        if self.isBot:
-            if (
-                self.config["BOT_HISTORY"]
-                and self.config["BOT_HISTORY"]["save"]
-            ):
-                if self.config["BOT_HISTORY"]["path"]:
-                    self.bot_recorder = RecorderBot(
-                        self.config["BOT_HISTORY"]["path"]
-                    )
-                else:
-                    raise ValueError(
-                        "Path to save conversation is not provided."
-                    )
-
-    def start_dialogue(self, user_fname=None, restart: bool = False):
+        if self.isBot and self.config.get("BOT_HISTORY", {}).get("save"):
+            path = self.config.get("BOT_HISTORY", {}).get("path")
+            if path:
+                self.bot_recorder = RecorderBot(path)
+            else:
+                raise ValueError("Path to save conversation is not provided.")
+
+    def start_dialogue(
+        self, user_fname: str = None, restart: bool = False
+    ) -> Union[
+        Tuple[str, DialogueOptions],
+        Tuple[str, Dict[str, Any], DialogueOptions],
```

Comment:
Should we change the order in this tuple like `str, DialogueOptions, Dict[str, Any]?
This way it is easier to identify what is what based on the returns statement.

Reply:
Done